Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -749,6 +775,43 @@ theorem isSplittingField_iSup {p : ι → K[X]}
 
 end Supremum
 
+section Tower
+
+variable (E)
+
+variable {K : Type*} [Field K] [Algebra F K] [Algebra E K] [IsScalarTower F E K]
+
+/-- If `K / E / F` is a field extension tower, `L` is an intermediate field of `K / F`, such that
+either `L / F` or `E / F` is algebraic, then `E(L) = E[L]`. -/
+theorem adjoin_toSubalgebra_of_isAlgebraic (L : IntermediateField F K)
+    (halg : Algebra.IsAlgebraic F E ∨ Algebra.IsAlgebraic F L) :
```

Comment:
Same as above, it's better to avoid `∨`.

Reply:
Note that the conclusion of this result is not symmetric in `E` and `L`, in fact one of them is an abstract field while another one is an intermediate field. If we want two separated results, then the same proof needs to be copied twice.

Reply:
Mmm, ok, it doesn't matter a lot.

Reply:
I think we can keep this version and add `isAlgebraic_{left|right}` versions as special cases of it. Is it OK?

Reply:
Yes, perfect. You can make it private maybe. Having the two versions nobody will ever use the one with the `\or`.

Reply:
It's expected that the `∨` version will be used in `LinearDisjoint`, though. 😂